sdk/messaging_servicebus/src/service_bus/mod.rs

```diff
@@ -0,0 +1,318 @@
+use azure_core::{
+    error::{Error, ErrorKind, ResultExt},
+    HttpClient,
+};
+use bytes::Bytes;
+use chrono::Duration;
+use http::{
+    header::{AUTHORIZATION, CONTENT_LENGTH},
+    method::Method,
+    Request, Response,
+};
+use ring::hmac;
+use std::{ops::Add, sync::Arc};
+use url::{
+    form_urlencoded::{self, Serializer},
+    Url,
+};
+
+mod client;
+pub use self::client::Client;
+
+/// Default duration for the SAS token in days — We might want to make this configurable at some point
+const DEFAULT_SAS_DURATION: i64 = 1;
+
+/// Prepares an HTTP request
+#[inline]
```

Comment:
I don't think this should be inlined (it's called pretty rarely). If the compiler thinks it better to inline, we can let it do so.

Reply:
done ✅